### PR TITLE
[SPARK-30691][SQL][DOC] Add a few main pages to SQL Reference

### PR DESCRIPTION
**Pull request by huaxingao**
### What changes were proposed in this pull request?
Add  a few main pages

### Why are the changes needed?
To make SQL Reference complete.

### Does this PR introduce any user-facing change?
Yes

image

image

image

image

image

image

image

image

### How was this patch tested?
Manually build and check

### CODE REVIEW

**Comment by huaxingao** [comment]
cc @srowen @dilipbiswal

File: docs/sql-ref-syntax.md [hunk 1]
```diff
@@ -19,7 +19,7 @@ license: |
   limitations under the License.
 ---
 
-Spark SQL is a Apache Spark's module for working with structured data.
+Spark SQL is an Apache Spark's module for working with structured data.
```
Comment (srowen) [inline docs/sql-ref-syntax.md]: Actually, just remove the 'a'. The copy/pasted text is elsewhere and has this problem; if there are any more occurrences you can fix them.

**Comment by dilipbiswal** [comment]
@huaxingao Thanks for doing this. LGTM

**Comment by huaxingao** [comment]
@dilipbiswal 
I removed the sql-ref-functions section from the menu because that whole section is not filled. I didn't remove the individual .md files, so it's easier for us to continue working on this later.

**Comment by huaxingao** [comment]
@srowen 
One more round please.

**Review by srowen — APPROVED** [review]

**Comment by srowen** [comment]
Merged to master

**Comment by huaxingao** [comment]
Thank you very much!! @srowen @dilipbiswal

File: docs/sql-ref-syntax.md [hunk 2]
```diff
@@ -19,7 +19,4 @@ license: |
   limitations under the License.
 ---
 
-Spark SQL is a Apache Spark's module for working with structured data.
-This guide is a reference for Structured Query Language (SQL) for Apache 
-Spark. This document describes the SQL constructs supported by Spark in detail
-along with usage examples when applicable.
+Spark SQL is Apache Spark's module for working with structured data. The SQL Syntax section describes the SQL syntax in detail along with usage examples when applicable.
```
Comment (gatorsmile) [inline docs/sql-ref-syntax.md]: @huaxingao can we add the hyperlink in this main page?

Just need to list the following links in this page

- Data Definition Statements
- Data Manipulation Statements
- Data Retrieval(Queries)
- Auxiliary Statements

File: docs/sql-ref-syntax-aux.md [hunk 3]
```diff
@@ -19,7 +19,11 @@ license: |
...
-Spark. This document describes the SQL constructs supported by Spark in detail
-along with usage examples when applicable.
+Besides the major SQL statements such as Data Definition Statements, Data Manipulation Statements and Data Retrieval Statements, Spark SQL also supports the following Auxiliary Statements:
+
+- [ANALYZE](sql-ref-syntax-aux-analyze.html)
+- [CACHE](sql-ref-syntax-aux-cache.html)
+- [DESCRIBE](sql-ref-syntax-aux-describe.html)
+- [SHOW](sql-ref-syntax-aux-show.html)
+- [CONFIGURATION MANAGEMENT](sql-ref-syntax-aux-conf-mgmt.html)
+- [RESOURCE MANAGEMENT](sql-ref-syntax-aux-resource-mgmt.html)
```
Comment (gatorsmile) [inline docs/sql-ref-syntax-aux.md]: These are not consistent with the names in the left sidebar. We should update the naming in the sidebar

image

**Comment by huaxingao** [comment]
@gatorsmile 
So you want the names in the links exactly match the sidebar, right? I will have a follow-up to change this. 


image

**Comment by huaxingao** [comment]
It is like this after the fix. I will submit a follow-up PR now.

image

**Pull request closed without merging**